sbi/utils/user_input_checks.py

```diff
@@ -220,6 +221,11 @@ def process_pytorch_prior(prior: Distribution) -> Tuple[Distribution, int, bool]
     # This will fail for float64 priors.
     check_prior_return_type(prior)
 
+    # Potentially required wrapper if the prior returns an additional sample dimension
+    # for `.log_prob()`.
+    if prior.log_prob(prior.sample((10,))).shape == (10, 1):
```

Comment:
pyright complains here because (10,1) is not torch.Size

Reply:
I would also suggest using `prior.sample().ndim` as above (`if prior.sample().ndim ==0` in line 200). I would change this clause to an `elif` and move it up as I think that makes things more readable.